primary/src/header_waiter.rs

```diff
@@ -334,7 +343,18 @@ impl<PublicKey: VerifyingKey> HeaderWaiter<PublicKey> {
                 self.pending.retain(|_, (r, _)| r > &mut gc_round);
                 self.batch_requests.retain(|_, r| r > &mut gc_round);
                 self.parent_requests.retain(|_, (r, _)| r > &mut gc_round);
+
+                self.metrics
+                    .gc_header_waiter_latency
+                    .with_label_values(&[&self.committee.epoch.to_string()])
+                    .observe(now.elapsed().as_secs_f64());
             }
+
+            // measure the pending elements
+            self.metrics
+                .pending_elements_header_waiter
+                .with_label_values(&[&self.committee.epoch.to_string()])
```

Comment:
Are you sure you want to use epoch as a label value?  Aren't epochs something that changes over time?  This will keep creating new time series, meaning that you cannot then easily correlate old epochs with new epochs.  Usually something that evolves over time is kept in the same time series.

Reply:
Labels must be bounded in scope and not unlimited.  Otherwise it will break Prometheus.
If the epoch is just a counter, then it could be exposed that way.

Reply:
I chose to use the echo as part of the metrics primarily for debugging reasons until we get good confidence on how things work (epoch is a fairly new thing). Having this as part of the metric it self it will make it easy to track how node's behaviour change - keep in mind that epochs change on the fly. I am all for removing epoch once we don't feel we need it anymore,

@allan-bailey we expect epoch to change once per day - I am aware of the limitations of the cardinality on the labels - so I would err on keeping as is before we remove it as we don't look on that many different values on the upcoming period.